### PR TITLE
ARXIVNG-1338 added support for journal reference submissions

### PR DESCRIPTION
**Pull request by erickpeirson**
I did this by adding a new ``CreateJREFSubmission`` command that inherits from ``CreateSubmission``. It works just like a replacement, but the version is not incremented. I also broke down and added a ``classic_type`` field to the ``Submission`` domain object. This is the simplest way to make sure we preserve classic integrations at this point. 

This includes changes from #40 and #41 , so those PRs should be reviewed and merged first.

Review if you are able. Looking for one approved review here, for the sake of extra eyes.

### CODE REVIEW

**Comment by JaimieMurdock** [comment]
You had mentioned waiting on the review for this PR? Making this note just to remind you to ping when everything is updated.

File: core/arxiv/submission/services/classic/__init__.py [hunk 1]
```diff
@@ -266,68 +329,52 @@ def store_event(event: Event, before: Optional[Submission],
...
-                                      document_id=document_id,
-                                      status=models.Submission.SUBMITTED,
-                                      doc_paper_id=before.arxiv_id,
-                                      version=after.version)
+            db_sb = _create_withdrawal(document_id, before.arxiv_id,
+                                       after.version, after)
+
         # Adding DOIs and citation information (so-called "journal reference")
         # also requires a new row. The version number is not incremented.
         elif before.published and type(event) in [SetDOI, SetJournalReference]:
```
Comment (erickpeirson) [inline core/arxiv/submission/services/classic/__init__.py]: @JaimieMurdock Here's where we catch jref-related events

File: core/arxiv/submission/services/classic/__init__.py [hunk 2]
```diff
@@ -234,8 +236,69 @@ def _load_document_id(paper_id: str, version: int) -> int:
...
+    db_sb = models.Submission(type=models.Submission.WITHDRAWAL,
+                              document_id=document_id,
+                              version=version)
+    db_sb.update_from_submission(submission)
+    db_sb.doc_paper_id = paper_id
+    db_sb.status = models.Submission.SUBMITTED
+    return db_sb
+
+
+def _create_jref(document_id: int, paper_id: str, version: int,
```
Comment (erickpeirson) [inline core/arxiv/submission/services/classic/__init__.py]: Here's how we update the database

File: core/arxiv/submission/tests/test_classic_integration.py [hunk 3]
```diff
@@ -445,6 +441,155 @@ def test_replacement(self):
         self.assertEqual(db_replacement.document.paper_id, '1901.00123')
 
 
+class TestJREFIntegration(TestCase):
```
Comment (erickpeirson) [inline core/arxiv/submission/tests/test_classic_integration.py]: Here are the tests that focus on the DB integration

File: core/arxiv/submission/domain/tests/test_events.py [hunk 4]
```diff
@@ -72,12 +72,104 @@ def test_create_submission_replacement(self):
...
+                         self.submission.metadata.msc_class)
+        self.assertEqual(replacement.metadata.acm_class,
+                         self.submission.metadata.acm_class)
+        self.assertEqual(replacement.metadata.doi,
+                         self.submission.metadata.doi)
+        self.assertEqual(replacement.metadata.journal_ref,
+                         self.submission.metadata.journal_ref)
+
+
+class TestDOIorJREFAfterPublish(TestCase):
```
Comment (erickpeirson) [inline core/arxiv/submission/domain/tests/test_events.py]: This just tests the events/commands themselves

**Comment by erickpeirson** [comment]
@JaimieMurdock This is probably ready to look at. Pardon the extra cruft, was cleaning up/refactoring some other bits here and there. Added some comments to highlight the substantive changes

**Pull request closed without merging**